dcap/types/src/quote3.rs

```diff
@@ -65,20 +74,33 @@ impl Error {
     }
 }
 
+// Using a from nom::Err that panics because of nom's generic implementation.
+// Without it all of the nom calls would need to specify the type like:
+// ```rust
+//  let (bytes, value) = le_u16::<_, nom::error::Error<&[u8]>>(bytes)
+//      .expect("Should have been prevented by previous size check.");
+// ```
+impl From<nom::Err<nom::error::Error<&[u8]>>> for Error {
+    fn from(_: nom::Err<nom::error::Error<&[u8]>>) -> Self {
+        panic!("Nom errors should be prevented by size checks when parsing a Quote3");
+    }
+}
+
```

Comment:
Thoughts on this?
I felt specifying the error type in each nom invocation was a bit heavy handed so left the error conversion to have it panic.

Reply:
Also if there is some rustism I'm missing to make this cleaner, I'm all for that.

Reply:
The problem is that this conversion is global, but the checks that make it OK are local. So, like, if someone later uses `Quote3Error` as their return type, and also use `nom` for something else, they can trigger a panic even though they may want a handled error.

IMO the best option here is to write a quick macro for this, e.g. `nom_expect!(tt)` that appends the `.expect("Nom errors...")`, and trust that LLVM will handle the static string properly.

Reply:
Added wrapper functions. I didn't look too far into the macro implementation but I think it would end up being fragile by needing to break up the function name and parens to insert the error type. 
It's possible to macro the 2 wrapper functions written, but it felt like the downside of obfuscation outweighed the just do it twice.